src/libraries/System.Net.HttpListener/src/System/Net/Windows/HttpListener.Windows.cs

```diff
@@ -451,6 +451,14 @@ private unsafe void CloseRequestQueueHandle()
                 if (NetEventSource.IsEnabled) NetEventSource.Info($"Dispose ThreadPoolBoundHandle: {_requestQueueBoundHandle}");
                 _requestQueueBoundHandle?.Dispose();
                 _requestQueueHandle.Dispose();
+                try
+                {
+                    Interop.Kernel32.CancelIoEx(_requestQueueHandle, null); // this cancels the synchronous call to HttpReceiveHttpRequest
+                }
+                catch (ObjectDisposedException)
+                {
+                    // ignore the exception since it only means that the queue handle has been successfully disposed
```

Comment:
It's not clear that this is correct. It looks like this exception is not an indicator of success, but will be thrown due to `_requestQueueHandle` being disposed when `CancelIoEx` is called -- in which case nothing will have been cancelled. Can you update the comment with more detail on the behavior you're expecting?

```suggestion
                    // Ignore the exception since it only means that the queue handle has been successfully disposed
```

Reply:
Same as other comment: found the discussion re: race condition, would just like some improved comments that make it clear what is expected here.